lib/http/response/body.rb

```diff
@@ -53,7 +53,10 @@ def to_s
           @streaming  = false
           @contents   = String.new("").force_encoding(encoding)
 
-          while (chunk = @stream.readpartial)
+          length = @length || Float::INFINITY
```

Comment:
> will merge as soon as it's ready :D before this weekend.

Famous last words :D 

I have a tentative commit here which I don't know if I'll include: I was going to make `Response#content_length` return `Float::INFINITY` when header not explicitly set. It's a bit avant-garde, since it's not completely correct, as 304 and 204 technically have content length 0. This could be arranged, though. This would make the method semantically more correct, but it's public API that could break some clients. Which could be ok, as it's v3, so breaking is acceptable if justified. 

I pass the potato to you.